core/services/workflows/engine.go

```diff
@@ -832,7 +832,10 @@ func NewEngine(cfg Config) (engine *Engine, err error) {
 
 	workflow.id = cfg.WorkflowID
 	workflow.owner = cfg.WorkflowOwner
-	workflow.name = cfg.WorkflowName
+	workflow.name = hex.EncodeToString([]byte(cfg.WorkflowName))
+	if len(workflow.name) > 20 { // TODO: require name to be exactly 10 bytes
+		workflow.name = workflow.name[:20]
```

Comment:
This only solves cases where the name is too long, but it could also be too short

Reply:
Also `20` -> `10`

Reply:
yeah, hence the TODO, I think this should be validated deeper, when validating a new spec.
Also it's 20 hex characters :) This is actually a bit confusing - do we require users to use only 10 ascii characters for the name or should they put a hex string? @DeividasK any preference?

Reply:
Leaving this up to the spec validator.